src/tcp_wrap.cc

```diff
@@ -113,7 +95,7 @@ void TCPWrap::Initialize(Local<Object> target,
 
   // Create FunctionTemplate for TCPConnectWrap.
   Local<FunctionTemplate> cwt =
-      FunctionTemplate::New(env->isolate(), NewTCPConnectWrap);
+      FunctionTemplate::New(env->isolate(), NewConnectWrap);
```

Comment:
Seeing how NewConnectWrap is just a CHECK one-liner, I'd be okay with replacing it with a C++11 lambda.  I'll leave that up to you.